packages/fxa-auth-server/lib/senders/emails/templates/newDeviceLogin/index.stories.ts

```diff
@@ -14,18 +14,130 @@ const createStory = storyWithProps(
   'newDeviceLogin',
   'Sent to notify the account that a new device has signed in.',
   {
-    ...MOCK_LOCATION,
     clientName: 'Firefox',
     passwordChangeLink: 'http://localhost:3030/settings/change_password',
     link: 'http://localhost:3030/settings',
   }
 );
 
+export const NewDeviceLoginFirefoxWithPrimary = createStory(
+  {
+    primaryEmail: 'primaryFoo@bar.com',
+  },
+  'New device login through Firefox - Missing Details - Primary email only'
+);
+
+export const NewDeviceLoginFirefoxWithDevice = createStory(
+  {
+    ...MOCK_DEVICE,
+  },
+  'New device login through Firefox - Missing Details - Device only'
+);
+
+export const NewDeviceLoginFirefoxWithPlace = createStory(
+  {
+    ...MOCK_PLACE,
+  },
+  'New device login through Firefox - Missing Details - Location only'
+);
+
+export const NewDeviceLoginFirefoxWithIp = createStory(
+  {
+    ...MOCK_IP,
+  },
+  'New device login through Firefox - Missing Details - IP only'
```

Comment:
I hate to push back on all the work you did here, but of my biggest questions as I'm looking at all of these new states is if some of these are actually feasible for users to see. Yes we have conditionals around each of these values in the partial, but will we ever have every other one of these missing, and _only_ have the IP, or date, or time, etc.? If I saw these states in Storybook, I think I would come to the conclusion that essentially two states are possible: full details, or a single detail, and I don't believe that's accurate.

If we check what we're sending into the [template values in `newDeviceLogin`](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/senders/email.js#L1398), for example, the possible states for the location partial aren't exactly what's laid out in Storybook here. The location partial should show `device`, `ip`, `location`, `date`, and `time`, but not `primaryEmail`. Unless it's possible for those values to be `null` or `undefined` (I wish we had Typescript here), I think that's the only possible state for this partial for that particular email, in which case we could just update the default Storybook for this email to show that.

I checked out what Bianca O. pointed out in Slack too that in the `verify` template, `.date` and `.time` are missing. If you check [what we're sending into the that template](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/senders/email.js#L627), that seems to be expected, and the only possible states that the location partial can be in for `verify` is no primary email, date, or time. To be fair, I have no idea why this is the case. It could have been a product decision not to include all the details into this partial in some of our emails, or, it could've been engineering oversight due to copy and pasting code without it.

I don't think it's going to be worth combing through every single email to see what we're sending down into the template. If/when we get #10601 done, we'll see more production-like values here anyway.

I think we should probably add this location template values dependency to the Confirmation Needed doc to confirm that we purposefully leave out some of this data sometimes.

It could be worth pulling this out and into its own "partials" section in Storybook though, and cover a few different states that can happen there. It would be way less lines added - even if all of the states laid out in this PR were valid, I would hesitate to r+ nearly 1200 new lines in the codebase when weighing how much value this brings us from a state documenting perspective vs maintenance burden. 😕 I think a few states that could be neat to show would be the various `location` states [that can be created from the `_constructLocationString` function](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/senders/email.js#L377).

Anyway sorry for such a long comment, I wrote this out to gather my thoughts about all of this too, let me know what your thoughts are. 😄

Reply:
Thanks for the feedback! As previously discussed, I am aligned with **not** r+ 1200 new lines in the codebase - it's definitely **not** the best approach.

I initially liked the idea of having a "partials" section in Storybook that displays all of the partials used in all templates, as well as their different states. However, I think the location partial is a special case as it may be the only partial that actually changes and is utilized among several email templates. While there are other partials with conditionals (e.g., payment plan details, payment provider, etc), their different states are already covered within their specific email templates in Storybook (e.g., subscriptionFirstInvoice/Discount, subscriptionSubsequentInvoice, etc). I believe they should stay within the email section as it's easier to keep track of the different email variations for a template. Yet if we keep them there, and also include them in the "partials" section, we would have extra unnecessary code.

Perhaps the email description should just be revised along the lines of "Sent to users to verify new login - information may vary depending on information available, such as primary email, device, location, IP, date and/or time".

Please let me know your thoughts when you have a moment. Thanks again!

Reply:
> However, I think the location partial is a special case as it may be the only partial that actually changes and is utilized among several email templates. 

This is true for now, but we also have a few others like `paymentProvider` used in a few templates that we could pull out and show a partials for, and might have others in the future. It might be confusing though on when to add one into the section and when not to, and I like that it's just "emails" at a top level, so I'm fine with not having a partials section. 👍

> Perhaps the email description should just be revised along the lines of "Sent to users to verify new login - information may vary depending on information available, such as primary email, device, location, IP, date and/or time".

Hmm, but we would need to do that for every single template using the location partial, and remember to add it for new ones. I'm still thinking it might be best to just add it to that confirmation needed doc and point out that we're not always sending in all of the info, because we might should be when we aren't. 🤔 Or, maybe we're purposefully not sending the location template values in for some reason, in which case we could document why.

Reply:
> 
> Hmm, but we would need to do that for every single template using the location partial, and remember to add it for new ones. I'm still thinking it might be best to just add it to that confirmation needed doc and point out that we're not always sending in all of the info, because we might should be when we aren't. 🤔 Or, maybe we're purposefully not sending the location template values in for some reason, in which case we could document why.

It sounds like these are our options: (1 - best case) get confirmation on a universal location partial that could be used in emails that require one, (2) revise the description of every single template using the location partial to let whoever is using Storybook know that information within the partial may vary, (3 - worst case) comb through every email with a location partial, note its variation of the location partial, update the location partial to include the variation, and then revise every email with the correct variant of the location partial. If we are not able to get a location partial that can be universally utilized, I would choose option 2 as this is a nice-to-have Storybook feature.

Please let me know if you're aligned with the above and/or if I'm missing something, and I'll add it to the Confirmation Needed doc. Also, should we push this issue to the next sprint? 👀

Reply:
I think we should go with option 1 for now. 👍

> Also, should we push this issue to the next sprint? 👀

I think instead, you can just change the title/description of the ticket to be around updating a few pieces of documentation for our emails. If/when we have a ticket regarding the partial, I think we can file it out of the confirmation needed meeting.